src/Logs.js

```diff
@@ -18,123 +22,161 @@ function Logs(props) {
       </div>
-    )
-  } else if (dayHasLog) {
-      return (
-        <div className="logScreenButtons">
-          <Button className="logScreenButton" onClick={() => props.currentScreen(<LogScreenRemove />)} currentScreen={props.currentScreen}>Remove Log</Button>
-          <Button className="logScreenButton" onClick={() => props.currentScreen(<LogScreenView />)}>View log</Button>
-        </div>
-  )
+    );
+  } if (dayHasLog) {
+    return (
+      <div className="logScreenButtons">
+        <Button className="logScreenButton" onClick={() => props.currentScreen(<LogScreenRemove />)} currentScreen={props.currentScreen}>Remove Log</Button>
+        <Button className="logScreenButton" onClick={() => props.currentScreen(<LogScreenView />)}>View log</Button>
+      </div>
+    );
   }
 }
 
-function LogScreenCreate(props) {
+function LogScreenCreate() {
   const [exerciseToAdd, setExerciseToAdd] = useState('');
   const [currentlySelectedExercise, setCurrentlySelectedExercise] = useState('');
   const [displayedExercises, setDisplayedExercises] = useState([]);
   const [exerciseIndex, setExerciseIndex] = useState(0);
   const [isDisplayed, setIsDisplayed] = useState('flex');
+  const [warningMessage, setWarningMessage] = useState();
   const [logStats, setLogStats] = useContext(LogStatsContext);
   const [clickedDay] = useContext(ClickedDayContext);
 
   function addExercise() {
-      setDisplayedExercises(displayedExercises.concat(exerciseToAdd));
-      setExerciseIndex(exerciseIndex + 1);
-      setLogStats({...logStats, [exerciseIndex]: {exerciseName: '', Sets: '', Reps: '', Weight: ''}})
+    const exercise = {
+      exerciseName: currentlySelectedExercise,
+      Sets: '',
+      Reps: '',
+      Weight: '',
+    };
+
+    setWarningMessage();
+    setDisplayedExercises(displayedExercises.concat(exerciseToAdd));
+    setExerciseIndex(logStats.length + 1);
+    setLogStats([...logStats, exercise]);
   }
 
   useEffect(() => {
-    setExerciseToAdd(<ExerciseRow exerciseName={currentlySelectedExercise} exerciseIndex={exerciseIndex} />)
-  }, [currentlySelectedExercise, exerciseIndex]);
+    setExerciseToAdd(<ExerciseRow exerciseName={currentlySelectedExercise} exerciseIndex={exerciseIndex} />);
+    setExerciseIndex(logStats.length);
+  }, [currentlySelectedExercise, exerciseIndex, logStats]);
 
   function handleChange(exerciseName) {
-    setCurrentlySelectedExercise(exerciseName)
+    setCurrentlySelectedExercise(exerciseName);
   }
 
   function handleSubmit(event) {
-    addLogToDatabase(logStats, moment(clickedDay).format('DD MM YYYY'));
-    setIsDisplayed('none')
+    let allInputsFilled = true;
+    let allInputsNumber = true;
+
+    for (let i = 0; i < logStats.length; i++) {
+      const exercise = logStats[i];
+
+      for (let j = 1; j < exercise.length; j++) {
+        const exerciseData = exercise[j];
+
+        if (isNaN(exerciseData[1]) || exerciseData[1] < 0) {
```

Comment:
Why are you using exerciseData[1] here. This is why using arrays everywhere gets confusing. There's no way to know what exerciseData[1] means